lib/core/dataAccess.js

```diff
@@ -125,70 +146,89 @@ class DataAccess {
    */
   async CheckTableColumnExist(tableName, columnName) {
     try {
-      const sql = `SELECT column_name FROM information_schema.columns WHERE table_name = '${tableName}' AND column_name = '${columnName}' LIMIT 1`
-      const res = await this.Transaction({ params: [{sql}] }) // v0.1.9 bug fixes
-      if (res[0].rows.length > 0) return true
-      return false
+      const sql = `SELECT column_name FROM information_schema.columns WHERE table_name = '${tableName}' AND column_name = '${columnName}' LIMIT 1`;
+      const res = await this.Transaction({ params: [{ sql }] }); // v0.1.9 bug fixes
+      return res.length > 0;
     } catch (e) {
-      throw e
+      throw e;
     }
   }
 
   /**
    * @description check whether where clause includes illegal operator, e.g. ===
    * @param {string} whereClause
+   * @param {bool} operatorRequired
    */
-  CheckWhereClauseOperator(whereClause) {
-    let illegal = true
+  CheckWhereClauseOperator(whereClause, operatorRequired = true) {
+    // The conditions for legality are opposite depending on whether or not an operator is required/allowed in the statement.
+    let illegal = operatorRequired;
     this.op.every(p => {
       if (
         String(whereClause)
           .toUpperCase()
-          .indexOf(p) !== -1 &&
+          .search(p) !== -1 &&
         String(whereClause)
           .toUpperCase()
           .split(p).length === 2
       ) {
-        illegal = false
-        return false
+        illegal = !operatorRequired;
+        return !operatorRequired;
       }
-      return true
-    })
-    if (illegal) throw new Error('condition operator is illegal')
+      return operatorRequired;
+    });
```

Comment:
I am not sure, if you wanna skip checking when `operatorRequired` is false, why dont you add a judgement before every loop?

Reply:
The goal isn't to skip checking. In a between statement: createdAt BETWEEN DATE(1/12019) AND DATE(2/2/2019), the right side of the AND cannot have an operator, so the operatorRequired variable allows the function to switch between requiring that there IS an operator or requiring that there IS NOT an operator.

Maybe this return should always be false since it's illegal?

Reply:
I'm not sure what you mean. The AND is checked by the the regex that splits on each of this.op, no?

Reply:
uhh, i see, in the `CheckWhereClause`. k, got it.